### ISSUE TITLE
ndimage.interpolation.shift differences with cupyx

### ISSUE THREAD

**Issue by chrisroat** [mritools/cupyimg#6]
Great package and great effort.  Thanks for doing this!

In leaving https://github.com/cupy/cupy/issues/4082, which affects both repos, I noticed some difference in behavior between the repos:

```
import cupy as cp
import cupyx.scipy.ndimage.interpolation as interp_x
import cupyimg.scipy.ndimage.interpolation as interp_img

# Expected
interp_x.shift(cp.zeros(10), 2, cval=-1)
array([-1., -1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.])

# Some strange interpolation happening?
interp_img.shift(cp.zeros(10), 2, cval=-1)
array([-1.        , -0.83333333, -0.16666667,  0.        ,  0.        ,
        0.        ,  0.        ,  0.        ,  0.        ,  0.        ])
```

I also note that cupyimg's version supports order>1, which cupyx does not.  Was that intentional?

My case is the desire to use cupyimg with order=3 to be in line with scipy itself, but I am concerned about the difference noted above.

**Comment by grlee77** [comment]
I am planning to contribute the spline orders 2-5 upstream to CuPy in the next month or so. Hopefully it will be available for CuPy 9.0.

See also https://github.com/mritools/cupyimg/issues/5#issuecomment-685926328 on future plans for this package in general. Since that comment was posted, the majority of remaining NumPy and SciPy functions here have been implemented upstream in CuPy, but there are a still a few left to go (including the spline interpolation support you noticed above).

That interpolation behavior does look like a bug! The intention was to have the behavior for `cupyimg` match the proposed behavior from scipy/scipy#12776 and then to later update CuPy to match that as well. Similarly `wrap` and `reflect` modes should work as their repaired versions in scipy/scipy#12767. Finally there is support for complex-valued inputs in this respository that is not currently in CuPy or SciPy (but hopefully will be in the future).

**Comment by chrisroat** [comment]
Just a clarification.  I didn't realize cupyimg had gone back to order=3 as the default, so the fair comparison is with scipy (since order=1 does not show the problem):

```
import cupy as cp
import numpy as np
import scipy.ndimage.interpolation as interp_sp
import cupyx.scipy.ndimage.interpolation as interp_x
import cupyimg.scipy.ndimage.interpolation as interp_img

interp_x.shift(cp.zeros(10), 2, order=1, cval=-1)
array([-1., -1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.])

interp_sp.shift(np.zeros(10), 2, order=1, cval=-1)
array([-1., -1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.])

interp_img.shift(cp.zeros(10), 2, order=1, cval=-1)
array([-1., -1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.])



interp_sp.shift(np.zeros(10), 2, order=3, cval=-1)
array([-1., -1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.])

interp_img.shift(cp.zeros(10), 2, order=3, cval=-1)
array([-1.        , -0.83333333, -0.16666667,  0.        ,  0.        ,
        0.        ,  0.        ,  0.        ,  0.        ,  0.        ])
```
